corebuild/integration/ILLink.Tasks/ILLink.Tasks.targets

```diff
@@ -226,15 +235,18 @@
 
   </Target>
 
-
   <!-- Computes the managed assemblies that are input to the
        linker. Includes managed assemblies from
        ResolvedAssembliesToPublish, and IntermediateAssembly. -->
   <Target Name="_ComputeManagedAssembliesToLink"
-          DependsOnTargets="_ComputeManagedResolvedAssembliesToPublish">
+          DependsOnTargets="_ComputeManagedResolvedAssembliesToPublish;_ComputePlatformLibraries">
     <ItemGroup>
       <_ManagedAssembliesToLink Include="@(IntermediateAssembly)" />
       <_ManagedAssembliesToLink Include="@(_ManagedResolvedAssembliesToPublish)" />
+      <_NetCoreFrameworkAssembliesToLink Include="@(PlatformLibraries)" />
+      <_NetCoreFrameworkAssembliesToLink Include="@(_ManagedResolvedAssembliesToPublish->WithMetadataValue('Filename', 'System.Private.CoreLib'))" />
+      <_ApplicationAssembliesToLink Include="@(_ManagedAssembliesToLink)" />
+      <_ApplicationAssembliesToLink Remove="@(_NetCoreFrameworkAssembliesToLink)" />
     </ItemGroup>
```

Comment:
Instead of computing separate itemgroups here, I would suggest adding metadata to ManagedAssembliesToLink with the assembly action. See my comment below for an explanation.